pkg/clustermesh/mcsapi/serviceexportsync.go

```diff
@@ -0,0 +1,150 @@
+// SPDX-License-Identifier: Apache-2.0
+// Copyright Authors of Cilium
+
+package mcsapi
+
+import (
+	"context"
+	"sync"
+
+	"github.com/cilium/hive/cell"
+	metav1 "k8s.io/apimachinery/pkg/apis/meta/v1"
+	mcsapiv1alpha1 "sigs.k8s.io/mcs-api/pkg/apis/v1alpha1"
+
+	"github.com/cilium/cilium/pkg/clustermesh/mcsapi/types"
+	"github.com/cilium/cilium/pkg/k8s/client"
+	"github.com/cilium/cilium/pkg/k8s/resource"
+	slim_corev1 "github.com/cilium/cilium/pkg/k8s/slim/k8s/api/core/v1"
+	"github.com/cilium/cilium/pkg/k8s/utils"
+	"github.com/cilium/cilium/pkg/kvstore"
+	"github.com/cilium/cilium/pkg/kvstore/store"
+	"github.com/cilium/cilium/pkg/logging"
+	"github.com/cilium/cilium/pkg/logging/logfields"
+)
+
+var log = logging.DefaultLogger.WithField(logfields.LogSubsys, "mcsapi")
+
+// ServiceExportResource builds the Resource[ServiceExport] object.
+func ServiceExportResource(lc cell.Lifecycle, cs client.Clientset, opts ...func(*metav1.ListOptions)) resource.Resource[*mcsapiv1alpha1.ServiceExport] {
+	if !cs.IsEnabled() {
+		return nil
+	}
+	lw := utils.ListerWatcherWithModifiers(
+		utils.ListerWatcherFromTyped(cs.MulticlusterV1alpha1().ServiceExports("")),
+		opts...,
+	)
+	return resource.New[*mcsapiv1alpha1.ServiceExport](lc, lw, resource.WithMetric("ServiceExport"))
+}
+
+type ServiceExportSyncParameters struct {
+	ClusterName    string
+	ServiceExports resource.Resource[*mcsapiv1alpha1.ServiceExport]
+	Services       resource.Resource[*slim_corev1.Service]
+	Backend        store.SyncStoreBackend
+	StoreFactory   store.Factory
+	Store          store.SyncStore
+	SyncCallback   func(context.Context)
+}
+
+func StartSynchronizingServiceExports(ctx context.Context, wg *sync.WaitGroup, cfg ServiceExportSyncParameters) {
```

Comment:
Nit. I wouldn't take the waitgroup as parameter, just create it in this function to ensure that when it returns all go routines terminated.